crates/store/re_protos/proto/rerun/v0/redap.proto

```diff
@@ -0,0 +1,275 @@
+// This makes it possible to directly connect to these services with the Rerun Viewer in minimal
+// deployments without the need to set up a separate catalog server.
+//
+// In such a minimal deployment, the exposed Catalog might consist of a single Collection while the Add/RemoveEntry
+// APIs are unimplemented.
+
+// The Catalog service is for listing and management of multiple catalog entries.
+service Catalog {
+    // List all the available entries in the Catalog
+    //
+    // This must be implemented by all Catalog-compatible services.
+    rpc ListEntries(ListEntriesRequest) returns (ListEntriesResponse) {}
+
+    // Add a new entry to the catalog
+    //
+    // This can be omitted by minimal/static catalogs.
+    rpc AddEntry(AddEntryRequest) returns (AddEntryResponse) {}
+
+    // Remove an entry from the catalog
+    //
+    // This can be omitted by minimal/static catalogs.
+    rpc RemoveEntry(RemoveEntryRequest) returns (RemoveEntryResponse) {}
+}
+
+// The Collection service is for managing and querying information about collections of resources.
+//
+// Because this service end-point is designed to be used in conjunction with a catalog, each request
+// include an optional `catalog_entry` field, which is used to route the request to the appropriate collection
+// within the catalog.
+//
+// Some catalog-providers may provide a default-collection, in which case the `catalog_entry` field may be omitted.
+service Collection {
+    // Get the Schema of the property table for a particular catalog entry
+    rpc GetPropertyTableSchema(GetPropertyTableSchemaRequest)
+        returns (GetPropertyTableSchemaResponse) {}
+
+    // Set the properties of one or more resources within a catalog entry
+    //
+    // May fail if the catalog entry is not writable.
+    rpc UpdateResourceProperties(UpdateResourcePropertiesRequest)
+        returns (UpdateResourcePropertiesResponse) {}
+
+    // Retrieve the table of properties for all resources stored in the collection.
+    //
+    // Possibly filtered / column projected.
+    //
+    // The schema of the returned Dataframe should match the results of `PropertyTableSchema`.
+    rpc GetResourceProperties(GetResourcePropertiesRequest)
+        returns (stream rerun.common.v0.DataframePart) {}
+
+    // Register one or more resources with catalog entry
+    //
+    // This API is only supported for some catalog entries.
+    rpc RegisterResources(RegisterResourcesRequest) returns (rerun.common.v0.DataframePart) {}
+
+    // Remove a resource from the a manually-managed catalog entry
+    //
+    // This API is only supported for some catalog entries.
+    rpc UnregisterResource(UnregisterResourcesRequest) returns (UnregisterResourceResponse) {}
+
+    // Remove all resources from a manually-managed catalog entry
+    //
+    // This API is only supported for some catalog entries.
+    rpc UnregisterAllResources(UnregisterAllResourcesRequest)
+        returns (UnregisterAllResourcesResponse) {}
```

Comment:
According to the "pizza" discussion, these should all move to the top-level, "catalog" scope. Instead, there should be `AddResource(resource_id)` and `RemoveResrouce(resource_id)`, which would just add/remove a pre-existing resource to the collection.